mono/utils/mono-tls.h

```diff
@@ -52,7 +52,32 @@ g_static_assert (TLS_KEY_DOMAIN == 0);
 #define mono_native_tls_alloc(key,destructor) ((*(key) = TlsAlloc ()) != TLS_OUT_OF_INDEXES && destructor == NULL)
 #define mono_native_tls_free TlsFree
 #define mono_native_tls_set_value TlsSetValue
-#define mono_native_tls_get_value TlsGetValue
+
+// TlsGetValue always writes 0 to LastError. Which causes problems. This never changes LastError.
+// See https://github.com/mono/mono/pull/15465.
+// See https://github.com/mono/mono/issues/15541.
+//
+// This is based on:
+//   mono_amd64_emit_tls_get https://github.com/mono/mono/blob/master/mono/mini/mini-amd64.c#L3784
+//   mono_x86_emit_tls_get   https://github.com/mono/mono/blob/master/mono/mini/mini-x86.c#L2201
+//
+#include <winternl.h>
+
+static inline
+void*
+mono_native_tls_get_value (unsigned index)
+{
+	PTEB const teb = NtCurrentTeb ();
+
+	g_assert (index < TLS_MINIMUM_AVAILABLE + 1024);
```

Comment:
I see you changed this... I assume `TLS_EXPANSION_SLOTS` was undefined so you replaced it with 1024. Perhaps `8 * sizeof(NtCurrentTeb()->Peb->TlsExpansionBitmapBits)` or something similar is an option?

Reply:
It is defined but only in obscure headers. It wasn't compiling.
winternal.h does not reveal this either.

Reply:
i.e. Peb->Bitmap.

Reply:
Let's remove it anyway, in case limit is raised.